### PR TITLE
feat: add workflow dropdown filter UI

### PR DESCRIPTION
**Pull request by Udit-takkar**
## What does this PR do?

Fixes https://linear.app/calcom/issue/CAL-1837/dropdown-ui

Screenshot 2023-06-09 at 12 08 17 AM

https://github.com/calcom/cal.com/assets/53316345/50f55dc1-8cad-4fa2-a657-2e594567205d

**Environment**: Staging(main branch) / Production

## Type of change

- New feature (non-breaking change which adds functionality)

### CODE REVIEW

File: packages/features/ee/workflows/components/WorkflowDropdownFilter.tsx [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+import { AnimatedPopover, Avatar } from "@calcom/ui";
+import { Layers, User, Check } from "@calcom/ui/components/icon";
+
+export const WorkflowDropdownFilter = () => {
+  const { t } = useLocale();
+  const session = useSession();
+
+  const [dropdownTitle, setDropdownTitle] = useState<string>(t("all_apps"));
+  const { data: query, pushItemToKey, removeItemByKeyAndValue, removeAllQueryParams } = useFilterQuery();
+  const { data: teams } = trpc.viewer.teams.list.useQuery();
```
Comment (Udit-takkar) [inline packages/features/ee/workflows/components/WorkflowDropdownFilter.tsx:19]: will change this query after sean's branch has been merged into feat/organizations

**Comment by CarinaWolli** [comment]
Another issue got created for that: https://github.com/calcom/cal.com/issues/9361 I think designs also got slightly updated

**Review by CarinaWolli — CHANGES_REQUESTED** [review]
Are you working on the functionality of the filter as well? 

There is a 0 next to the dropdown:
Screenshot 2023-06-09 at 10 39 21

Checking boxes doesn't really work as it should:
- I can't uncheck a team, once checked 
- Team unchecks when checking 'Yours'
- Unchecking 'Yours' also unchecks teams

**Comment by Udit-takkar** [comment]
@CarinaWolli  1) Yeah i am going to add the functionality of both event type filter and workflow filter in my next PR. This PR just adds UI. I will change title of dropdown when multiple options are selected like "Selected(2)" later after confirming with Ciaran.

2) Fixed the things you mentioned.  

https://github.com/calcom/cal.com/assets/53316345/ef0188c1-40dc-4f17-9040-63606c9b90e6

**Comment by Udit-takkar** [comment]
closing in favour of https://github.com/calcom/cal.com/pull/9417/files#r1223039889. I will make the changes in  that branch

**Pull request closed without merging**